src/fswAlgorithms/attGuidance/prescribedRot1DOF/prescribedRot1DOF.c

```diff
@@ -0,0 +1,189 @@
+#include <stdbool.h>
+#include <math.h>
+#include "architecture/utilities/linearAlgebra.h"
+#include "architecture/utilities/rigidBodyKinematics.h"
+
+/*! This method initializes the output messages and other parameters for this module.
+ @return void
+ @param configData The configuration data associated with this module
+ @param moduleID The module identifier
+ */
+void SelfInit_prescribedRot1DOF(PrescribedRot1DOFConfig *configData, int64_t moduleID)
+{
+    // Initialize the output messages
+    PrescribedMotionMsg_C_init(&configData->prescribedMotionOutMsg);
+    HingedRigidBodyMsg_C_init(&configData->spinningBodyOutMsg);
+
+    // Set the initial time
+    configData->tInit = 0.0;
+
+    // Set the initial convergence to true to enter the required loop in Update_prescribedRot1DOF() on the first pass
+    configData->convergence = true;
+
+}
+
+
+/*! This method performs a complete reset of the module. The input messages are checked to ensure they are linked.
+ @return void
+ @param configData The configuration data associated with the module
+ @param callTime [ns] Time the method is called
+ @param moduleID The module identifier
+*/
+void Reset_prescribedRot1DOF(PrescribedRot1DOFConfig *configData, uint64_t callTime, int64_t moduleID)
+{
+    // Check if the required input message is linked
+    if (!HingedRigidBodyMsg_C_isLinked(&configData->spinningBodyInMsg))
+    {
+        _bskLog(configData->bskLogger, BSK_ERROR, "Error: prescribedRot1DOF.spinningBodyInMsg wasn't connected.");
+    }
+}
+
+
+/*! This method profiles the prescribed trajectory and updates the prescribed states as a function of time.
+The prescribed states are then written to the output message.
+ @return void
+ @param configData The configuration data associated with the module
+ @param callTime [ns] Time the method is called
+ @param moduleID The module identifier
+*/
+void Update_prescribedRot1DOF(PrescribedRot1DOFConfig *configData, uint64_t callTime, int64_t moduleID)
+{
+    // Create the buffer messages
+    HingedRigidBodyMsgPayload spinningBodyIn;
+    HingedRigidBodyMsgPayload spinningBodyOut;
+    PrescribedMotionMsgPayload prescribedMotionOut;
+
+    // Zero the output messages
+    spinningBodyOut = HingedRigidBodyMsg_C_zeroMsgPayload();
+    prescribedMotionOut = PrescribedMotionMsg_C_zeroMsgPayload();
+
+    // Read the input message
+    spinningBodyIn = HingedRigidBodyMsg_C_zeroMsgPayload();
+    if (HingedRigidBodyMsg_C_isWritten(&configData->spinningBodyInMsg))
+    {
+        spinningBodyIn = HingedRigidBodyMsg_C_read(&configData->spinningBodyInMsg);
+    }
+
+    /* This loop is entered (a) initially and (b) when each attitude maneuver is complete. The reference angle is updated
+    even if a new message is not written */
+    if (HingedRigidBodyMsg_C_timeWritten(&configData->spinningBodyInMsg) <= callTime && configData->convergence)
+    {
+        // Store the initial time as the current simulation time
+        configData->tInit = callTime*1e-9;
```

Comment:
use the `NANO2SEC` macro to convert from nanoseconds to seconds.

Reply:
Done